clientlibs/java/src/main/java/org/upgradeplatform/responsebeans/Assignment.java

```diff
@@ -0,0 +1,114 @@
+package org.upgradeplatform.responsebeans;
+
+import java.util.Map;
+
+import org.upgradeplatform.client.ExperimentClient;
+import org.upgradeplatform.interfaces.ResponseCallback;
+import org.upgradeplatform.requestbeans.MarkExperimentRequestData;
+import org.upgradeplatform.utils.Utils.ExperimentType;
+import org.upgradeplatform.utils.Utils.MarkedDecisionPointStatus;
+
+public class Assignment {
+
+    private String target;
+    private String site;
+    private ExperimentType experimentType;
+    private Condition assignedCondition;
+    private Map<String, Factor> assignedFactor;
+
+    public Assignment(String target, String site, ExperimentType experimentType, Condition assignedCondition) {
```

Comment:
what if the constructor could take client as an argument? So you don't have to pass it into `markExperimentPoint` at the point of calling?

Reply:
i like that idea!